controller/engine_controller.go

```diff
@@ -1235,6 +1236,23 @@ func (m *EngineMonitor) restoreBackup(engine *longhorn.Engine, rsMap map[string]
 		"requestedRestoredBackupName": engine.Spec.RequestedBackupRestore,
 		"lastRestoredBackupName":      engine.Status.LastRestoredBackup,
 	})
+
+	// check if backup exists
+	backupURL := backupstore.EncodeBackupURL(engine.Spec.RequestedBackupRestore, engine.Spec.BackupVolume, backupTargetClient.URL)
+	backupInfo, err := backupTargetClient.BackupGet(backupURL, backupTargetClient.Credential)
+	if err != nil {
+		mlog.WithError(err).Error("failed to inspect backup config")
+		return errors.Wrapf(err, "cannot inspect backup %v config for backup restoration of engine %v", engine.Spec.RequestedBackupRestore, engine.Name)
+	}
+	if backupInfo == nil {
+		errMsg := "Cannot find backup for backup restoration of engine"
```

Comment:
Suggest the message to`remote backup not found` which will be used for the status.

Reply:
Modified. Thanks.